plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -127,6 +128,10 @@ def set_replica_target
       template[:spec][:replicas] = @doc.replica_target if template[:kind] == 'Deployment'
     end
 
+    def set_name
+      template[:metadata][:name] = @doc.kubernetes_role.resource_name
```

Comment:
this is about metadata.name, not deployment.spec.template.metadata.name

atm if you have 2 projects that both have metadata.name=appserver or
something like that they will override each other ... so we need to enforce
some kind of uniqueness to make sure that does not happen

On Wed, Jun 29, 2016 at 9:31 AM, rata <notifications@github.com> wrote:

> In plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb
> <https://github.com/zendesk/samson/pull/1110#discussion_r68980993>:
>
> > @@ -127,6 +128,10 @@ def set_replica_target
> >        template[:spec][:replicas] = @doc.replica_target if template[:kind] == 'Deployment'
> >      end
> >
> > +    def set_name
> > +      template[:metadata][:name] = @doc.kubernetes_role.resource_name
>
> And I'm kind of lost of why this is needed. It doesn't seem nice if we
> change what the user already specified.
>
> If deployment.spec.template.metadata.name is not specified (is not
> mandatory, in fact I don't have it in any of my projects :)), kubernetes
> will use something for us. And if the user specified it, I guess he/she
> might have a reason.
>
> Why do we want to set this to something in particular? Sorry, I don't
> understand :-/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1110/files/69e70907f8826375ada24e525c7a1777046671ee#r68980993>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAsZ1r961hLgbdTlm3CZkxWmwlhDh_Xks5qQp33gaJpZM4JAoD7>
> .
>